docs/pair_hmm.tex

```diff
@@ -0,0 +1,101 @@
+\documentclass[nofootinbib,amssymb,amsmath]{revtex4}
+\usepackage{mathtools}
+\usepackage{amsthm}
+\usepackage{algorithm}
+\usepackage{algpseudocode}
+\usepackage{lmodern}
+\usepackage{graphicx}
+\usepackage{color}
+
+%Put an averaged random variable between brackets
+\newcommand{\ave}[1]{\left\langle #1 \right\rangle}
+\newcommand{\HC}{\texttt{HaplotypeCaller}}
+\newcommand{\Mutect}{\texttt{Mutect}}
+\newcommand{\code}[1]{\texttt{#1}}
+\newcommand{\mc}[1]{\mathcal{#1}}
+
+
+\newtheorem{lemma}{Lemma}
+\newtheorem{corollary}{Corollary}
+
+\def\SL#1{{\color [rgb]{0,0,0.8} [SL: #1]}}
+\def\DB#1{{\color [rgb]{0,0.8,0} [DB: #1]}}
+
+\begin{document}
+
+\title{Pair HMM probabilistic realignment in HaplotypeCaller and Mutect}
+\author{David Benjamin\footnote{The author took no part in development of the methods described below -- credit belongs to several others on the GATK team. }}
+\email{davidben@broadinstitute.org}
+\affiliation{Broad Institute, 75 Ames Street, Cambridge, MA 02142}
+
+\date{\today}
+
+\begin{abstract}
+After generating candidate haplotypes, the GATK tools \HC~and \Mutect~realign reads against these haplotypes to obtain a matrix of likelihoods for each read to be derived from each haplotype.  Here we describe the probabilistic model specifying this likelihood as well as its computational implementation.  We do not describe the translation of this implementation into native code optimized for vectorized architectures.
+\end{abstract}
+
+\maketitle
+
+\section{The Pair HMM model}
+We want to calculate the probability $P( \mc{R} | \mc{H})$ of read $\mc{R}$ to be sequenced from haplotype $\mc{H}$, where the haplotypes are sufficiently long that reads are contained within them.  This likelihood is the sum of likelihoods of all possible alignments $\mc{A}$ of $\mc{R}$ to $\mc{H}$:
```

Comment:
It's certainly not substantial evidence, no matter the prior.  But did you have any thoughts you wanted to add on why we sum?

Reply:
Summing over alignments is marginalizing over all possible values of a latent variable.  If we believe the Pair HMM model then it's the mathematically correct thing to do and our hand is forced.  In the IGV screenshot above, if we didn't marginalize but instead including only the likeliest alignment, the bottom gray read would still be in an unsettling state, a tie between the ref and the deletion.  It's possible that marginalizing breaks the tie in favor of the deletion, but this is a perturbation.

From the point of view of likelihoods, however, this seems correct to me, because likelihoods (as opposed to posterior probabilities) answer the question: "*if* this allele is in fact present, does this read make sense?"  Given that the read is a perfect match to both haplotypes, the likelihoods should be basically equal.